### ISSUE TITLE
Upgrade from 1.11 to 1.12.1 - AKS cluster changes stuck in error loop

### ISSUE THREAD

**Issue by dtzar** [kubernetes-sigs/cluster-api-provider-azure#4388]
/kind bug

**What steps did you take and what happened:**
Have an existing AKS cluster 1.26.6 under management using CAPZ 1.11.6, upgrade to CAPZ 1.12.1, make a change to upgrade to kubernetes 1.27.7.  Once I do this there are no perceived errors, but the upgrade does not complete.  Seemingly no relevant errors on the capz management controller log.  On the ASO controller log I see the following in a continuous loop:

```
I1214 18:42:36.599381       1 common.go:126] "msg"="applying ownership" "logger"="controllers.ManagedClustersAgentPoolController" "name"="pool1" "namespace"="default"
I1214 18:42:36.599466       1 common.go:73] "msg"="Owner does not yet exist" "NamespacedName"={"Namespace":"default","Name":"ascapz"} "logger"="controllers.ManagedClustersAgentPoolController" "name"="pool1" "namespace"="default"
E1214 18:42:36.599507       1 generic_reconciler.go:180] "msg"="Error claiming resource" "error"="Reason: WaitingForOwner, Severity: Warning, RetryClassification: RetryFast, Cause: Owner \"ascapz, Group/Kind: containerservice.azure.com/ManagedCluster\" cannot be found. Progress is blocked until the owner is created." "logger"="controllers.ManagedClustersAgentPoolController" "name"="pool1" "namespace"="default"
E1214 18:42:36.599535       1 generic_reconciler.go:361] "msg"="Encountered error impacting Ready condition" "error"="Reason: WaitingForOwner, Severity: Warning, RetryClassification: RetryFast, Cause: Owner \"ascapz, Group/Kind: containerservice.azure.com/ManagedCluster\" cannot be found. Progress is blocked until the owner is created." "logger"="controllers.ManagedClustersAgentPoolController" "name"="pool1" "namespace"="default"
I1214 18:42:36.617113       1 generic_reconciler.go:96] "msg"="Encountered error, re-queuing..." "logger"="controllers.ManagedClustersAgentPoolController" "name"="pool1" "namespace"="default" "result"={"Requeue":false,"RequeueAfter":30000000000}

```

**What did you expect to happen:**
The cluster applied the changes (upgraded).

**Environment:**

CAPZ Management cluster:
- cluster-api-provider-azure version: 1.12.1
- Kubernetes version: (use `kubectl version`): 1.27.3
- OS (e.g. from `/etc/os-release`): kind 0.20.0 using workload identity

**Comment by dtzar** [comment]
Potentially related to #4385

**Comment by nojnhuh** [comment]
I just tried this and don't see the same behavior. The only differences being upgrading CAPZ from v1.11.7 and upgrading the cluster from v1.28.0 to v1.28.3, but I don't expect any of that to make a difference here.

When and how exactly did you perform the Kubernetes upgrade to v1.27.7? In my test I was waiting to make sure the workload cluster was fully settled after the CAPZ upgrade before the Kubernetes upgrade.

And can you see that the ASO ManagedCluster really doesn't exist (`kubectl get managedclusters`)?

**Comment by nojnhuh** [comment]
I just tried this and don't see the same behavior. The only differences being starting from CAPZ v1.11.7 instead of v1.11.6 and upgrading form Kubernetes v1.28.0 to v1.28.3, but I wouldn't expect any of that to make a difference here.

When and how exactly did you perform the Kubernetes upgrade? I was waiting for things to settle down completely after the upgrade before trying that. Not sure if trying that in the middle of a CAPZ upgrade might affect things between the CAPZ restart, ASO restart, CRDs getting installed, CAPZ initial reconcile after restart.

And can you see that the ASO ManagedCluster really doesn't exist (`kubectl get managedclusters`)? And you're sure there aren't any CAPZ controller log errors?

**Comment by dtzar** [comment]
I figured out the problem.  The AzureManagedControlPlane was having a problem being created because I tried to update the definition for both the AzureManagedControlPlane AND the node pool versions at the same time.  Logically you'd think it would just figure that out but it didn't for whatever the reason.  When I bumped only the AzureManagedControlPlane version, I saw this work and oddly in the portal the UX made it seem like it would upgrade the nodepools as well, but it didn't (good it's respecting my CAPZ definition :) )
![image](https://github.com/kubernetes-sigs/cluster-api-provider-azure/assets/3578906/e494bafb-88f6-4b00-b649-b6ab8cb24780)
 
 After this completed, I bumped the AzureManagedMachinePool versions and then those got upgraded without any issues.
